mtest/libmscore/all_elements/CMakeLists.txt

```diff
@@ -14,3 +14,6 @@ set(TARGET tst_layout_elements)
 
 include(${PROJECT_SOURCE_DIR}/mtest/cmake.inc)
 
+set(TARGET tst_tree_model)
+
+include(${PROJECT_SOURCE_DIR}/mtest/cmake.inc)
```

Comment:
Including `cmake.inc` causes the crash on AppVeyor due to the line [`include(FindStaticLibrary)`](https://github.com/musescore/MuseScore/blob/84d97c6ace691286c84db3a8a2d845563a4019ee/mtest/cmake.inc#L32), which loads [`build/FindStaticLibrary.cmake`](https://github.com/musescore/MuseScore/blob/master/build/FindStaticLibrary.cmake). That file creates a target called `zlibstat`. You cannot create two targets with the same name in the same directory, hence it fails the second time.

To avoid this, you can do something like this in `cmake.inc`:

```cmake
if (NOT ALREADY_BEEN_HERE)
    set(ALREADY_BEEN_HERE TRUE)
    include(FindStaticLibrary)
endif()
```

Or you can make use of CMake's [`include_guard(DIRECTORY)`](https://cmake.org/cmake/help/latest/command/include_guard.html) statement, but that would require a bit of restructuring as the `include_guard` must come after all code that needs to execute both times the file is run.

However, it should be enough to move the line `include(FindStaticLibrary)` to [`mtest/CMakeLists.txt`](https://github.com/musescore/MuseScore/blob/master/mtest/CMakeLists.txt) as that file is only run once in total, not once per test. (Unless @AntonioBL knows reason why it can't go there.)

Finally, you could just put the code that runs once per directory in a new file `inc_dir.cmake` that is only mentioned once in each directory's CMakeList.txt rather than once per test. (If you do this then please rename `cmake.inc` to `inc.cmake` so that we get proper syntax highlighting.)

---

Once the above is fixed, there is likely to be another error further down at the line [`DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/styles`](https://github.com/musescore/MuseScore/blob/84d97c6ace691286c84db3a8a2d845563a4019ee/mtest/cmake.inc#L140), because the files mentioned in the lines above will have already been installed at that location first time around. This can be fixed in the same way as before, with `if`, `include_guard` or moving to a file that is only executed once.

Reply:
Solution 3 is easy and most elegant, but it is not guaranteed to work.

Solutions 1 and 2 are a bit hacky, but require least effort and are guaranteed to work.

Solutions 4 and 5 require modifying the `CMakeList.txt` file in all test directories, which is quite a bit of effort. However, they are more elegant than 1 and 2 so it's a pity it wasn't done this way from the beginning. Don't bother with it now though.

Reply:
I put that line there because it was setting zlibstat as target link library in the next instruction; it worked because single CmakeList.txt files were defining one single test target. But I think that that line can be moved to `mtest/CMakeLists.txt` without problems.

Reply:
@AntonioBL, make sense. What about the `install` at the bottom of the file? The [`DESTINATION` is `${CMAKE_CURRENT_BINARY_DIR}/styles`](https://github.com/musescore/MuseScore/blob/84d97c6ace691286c84db3a8a2d845563a4019ee/mtest/cmake.inc#L140), where `CMAKE_CURRENT_BINARY_DIR` is the current test directory (inside the project build directory). Can the install be moved to `mtest/CMakeLists.txt` (where `CMAKE_CURRENT_BINARY_DIR` is the `mtest` directory) or are those files really needed in each test directory?

Reply:
In this case I put on purpose the destination dir as each single test dir since the test executables were searching for that directory relative to their own path, rather than relative to the parent application mtest.